pkg/agent/run.go

```diff
@@ -148,7 +144,7 @@ func Run(ctx context.Context, cfg cmds.Agent) error {
 		return err
 	}
 
-	proxy, err := proxy.NewAPIProxy(!cfg.DisableLoadBalancer, agentDir, cfg.ServerURL)
```

Comment:
Is this undoing something that was done for the recent data-dir adjustment(s)?

Reply:
Specifically, `agentDir` :arrow_right:  `cfg.DataDir`

Reply:
so to explain this change I want to explain how kubeconfigs for components (kubelet and kubeproxy) are initialized, so when the agent runs it use (lb.SupervisorURL) for the api address, which is 127.0.0.1 for the host and random port that lb starts with, what I did is making sure that this port is fixed and set to a specified number from the start which is 6444, this because this same port is used to initialize admin kubeconfig https://github.com/k3s-io/k3s/pull/2900/files/5d6ba53fcd075002b783354c7449de52c518e677#diff-750227e174e9fe230c0aede7c189127dff9b587b03449ecf875002d740dd59e7R145 which I use to set the labels on the etcd node

Reply:
oh sorry this must have gotten changed because of 1.19 changes, I will fix that

Reply:
fixed